src/manifest/mod.rs

```diff
@@ -141,39 +141,45 @@ struct CrateInformation {
 
 impl Crate {
     /// Returns latest wasm-pack version
-    pub fn return_wasm_pack_latest_version() -> Option<String> {
+    pub fn return_wasm_pack_latest_version() -> Result<Option<String>, failure::Error> {
         let current_time = chrono::offset::Local::now();
+        let old_metadata_file = Self::return_wasm_pack_file();
 
-        Self::return_wasm_pack_file()
-            .and_then(|contents| {
-                let last_updated = Self::return_stamp_file_value(&contents, "created")
+        match old_metadata_file {
+            Some(ref file_contents) => {
+                let last_updated = Self::return_stamp_file_value(&file_contents, "created")
                     .and_then(|t| DateTime::parse_from_str(t.as_str(), "%+").ok());
 
-                Self::return_stamp_file_value(&contents, "version").and_then(|v| {
-                    last_updated.and_then(|last_updated| {
+                last_updated
+                    .map(|last_updated| {
                         if current_time.signed_duration_since(last_updated).num_hours() > 24 {
-                            Self::return_api_call_result(current_time)
+                            Self::return_api_call_result(current_time).map(Some)
                         } else {
-                            Some(v)
+                            Ok(Self::return_stamp_file_value(&file_contents, "version"))
                         }
                     })
-                })
-            })
-            .map_or(Self::return_api_call_result(current_time), |value| {
-                Some(value)
-            })
+                    .unwrap_or_else(|| Ok(None))
+            }
+            None => Self::return_api_call_result(current_time).map(Some),
+        }
     }
 
-    fn return_api_call_result(current_time: DateTime<offset::Local>) -> Option<String> {
-        Self::return_latest_wasm_pack_version().and_then(|v| {
-            Self::override_stamp_file(current_time, &v).ok();
-            Some(v)
-        })
+    fn return_api_call_result(
+        current_time: DateTime<offset::Local>,
+    ) -> Result<String, failure::Error> {
+        let version = Self::return_latest_wasm_pack_version();
+
+        match version {
+            Ok(ref version) => Self::override_stamp_file(current_time, Some(&version)).ok(),
+            Err(_) => Self::override_stamp_file(current_time, None).ok(),
```

Comment:
Probably worth a comment here explaining why we override the time stamp file even when getting the version fails.

Reply:
Absolutely!